### DISCUSSION TITLE
Something overwrites padding of TabControl

### DISCUSSION THREAD

**Discussion by smalduck** [AvaloniaUI/Avalonia#9786] [Q&A]
If I create a TabControl like this in XAML

```xaml
<TabControl>
        <TabControl.Styles>
            <Style Selector="TabControl">
                <Setter Property="Padding" Value="0"/>
            </Style>
        </TabControl.Styles>

        <!-- Content -->

</TabControl>
```

it still somehow has padding. But if I define it like this

```xaml
<TabControl Padding="0">

        <!-- Content -->

</TabControl>
```

there is no style override. I looked through the docs but haven't found anything that overwrites the padding property on the TabControl yet. It's not a huge deal but it is a little weird. Can anybody help or give me a hint?

Thank you for reading through this!

**Comment by ahopper** [comment]
styles have scope and so your style will apply to any tabcontrol inside your tabcontrol but not the tabcontrol containing it or any other tabcontrols in your app.

**Reply by cyraid** [reply]
I've been able to style everything else as well except padding for SimpleTheme. Did you get this figured out @ghostcrafter551 ?

**Reply by smalduck** [reply]
Unfortunately I haven't found a solution yet.

**Reply by cyraid** [reply]
@ahopper In every scenario, in separate style sheet in Application.Styles, StyleInclude, TabControl.Styles, etc. etc. I cannot override padding. Even with different scopes. What way are you using to override padding?

**Reply by timunie** [reply]
@cyraid which I really wonder about this. IMO that should work. Can you give me more info? 
- Which Avalonia version?
- Which Theme do you use? 
- Any other TabControl styles defined somewhere?

**Reply by cyraid** [reply]
I gave up Avalonia and decided to use a different solution. But to answer your questions, I was using preview5, simpletheme, and didn't have tabcontrol styles anywhere else. Should be easy enough to reproduce with that info if you wanna track it down. But anyway, good luck to others. :)

**Reply by smalduck** [reply]
Now I'm interested. What do you use instead? I also don't develop in Avalonia any more because it wasn't the right choice for me personally. Do you have a good recommendation?

**Reply by timunie** [reply]
@cyraid @ghostcrafter551 I'm really sad you gave up Avalonia, but I also don't blame you and whish you the best. 

Anyway, thank you for the feedback and also the details shared. I'll at least take a look if I can reproduce your issue.

**Comment by timunie** [comment]
I can reproduce the issue in 11.0 preview5 and also latest nightly builds. I think this is due to nested Styles in the simple control theme for the TabControl. 

@maxkatz6 should I convert this into an issue?

**Comment by mysteryx93** [comment]
Note that with [MediaPlayer.Avalonia](https://github.com/mysteryx93/MediaPlayerUI.NET), when using `FluentTheme`, "something overrides something" and there are glitches with the seek bar when pressing. With `SimpleTheme` or `FluentAvalonia`, no such thing happens. Never could figure out where the bogus style would come from; it shouldn't be taking anything from the default theme to begin with.

Perhaps this is related?
